### PR TITLE
#11972: Add parameter generation step to sweeps CI workflow, use new tagging feature, and schedule nightly

### PR DESCRIPTION
**Pull request by jdesousa-TT**
### Ticket
#11972 

### Problem description
It was possible for vectors to become out of date with the files in main, meaning some tests could possibly break on CI runs if developers were concurrently working on changes.

### What's changed
Added parameter generation step to CI sweeps workflow using the new tagging functionality introduced in #11970.
Schedule this new run nightly.

### Checklist
- [ ] Post commit CI passes
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

**Review by xanderchin — APPROVED** [review]

File: .github/workflows/ttnn-run-sweeps.yaml
Comment (ttmchiou) [inline .github/workflows/ttnn-run-sweeps.yaml:23]: add "in-service" label here
or else you could pick up a out of commission builder machine

Comment (ttmchiou) [reply .github/workflows/ttnn-run-sweeps.yaml:23]: As a side-note, is this suppose to run on our CPU only builders? or does this need a card?
otherwise this will need
for grayskull
runs-on: [E150, in-service, cloud-virtual-machine]
for wormhole:
runs-on: [N150, in-service, cloud-virtual-machine]
runs-on: [N300, in-service, cloud-virtual-machine]

Comment (jdesousa-TT) [reply .github/workflows/ttnn-run-sweeps.yaml:23]: Just CPU for this job. Next job in the workflow needs hardware. Can you double check that's correct?

Comment (jdesousa-TT) [reply .github/workflows/ttnn-run-sweeps.yaml:23]: It worked when I ran this step before. Just runs-on: ${{matrix.arch}}

Comment (ttmchiou) [reply .github/workflows/ttnn-run-sweeps.yaml:23]: Can you post a link to the run with the new changes? I don't see any CI job on jdesousa/sweeps-ci-schedule.

if you reference the build/artifact.yaml file. you can see matrix.arch is defined in there, but no arch is defined in this workflow?

For reference, ripped from ./.github/workflows/build-artifact.yaml
```
build-artifact:
    strategy:
      matrix:
        arch: ${{ fromJson(inputs.arch || '["grayskull", "wormhole_b0", "blackhole"]') }}
```

File: .github/workflows/ttnn-run-sweeps.yaml
Comment (ttmchiou) [inline .github/workflows/ttnn-run-sweeps.yaml:78]: Does this timeout need to be this long?
Not sure if tests need to be 12 hours long.

Comment (jdesousa-TT) [reply .github/workflows/ttnn-run-sweeps.yaml:78]: I did a run of it last night, and even at 12 hours it timed out eventually. There is a lot of tests, and a lot of them have hangs which require a reset which takes a long time. I was talking with @xanderchin and @tt-rkim we will need to do some sort of parallelization of the tests across runners. I thought 12 hours might be enough for now but even already it's too short.
https://github.com/tenstorrent/tt-metal/actions/runs/10603188781

Comment (ttmchiou) [reply .github/workflows/ttnn-run-sweeps.yaml:75]: So what is the next steps for this PR?
Is there anymore work to be done? ie parallelization of runs?
It doesn't look like tests will pass in its current state?

Comment (jdesousa-TT) [reply .github/workflows/ttnn-run-sweeps.yaml:75]: Most of the time being taken right now is because @sjameelTT has added some large suites for concat that all hang. (#12073) He is going to add some asserts to catch a lot of these hanging cases which should shorten this significantly (without these tests it takes 3-4 hours rn). Parallelization is going to come after, but I think we can merge this to actually have the pipeline working but maybe time out for the time being. Not sure if you want to wait for #12073 to be completed or not before we merge this?

**Comment by jdesousa-TT** [comment]
Separating these changes into a few atomic PRs for readability.

**Pull request closed without merging**
